### PR TITLE
IE8 compatibility

### PR DESCRIPTION
**Pull request by e1ectronic**
Small changes to make DOMVM compatible with the Internet Explorer 8.

### CODE REVIEW

File: src/view/patchEvent.js [hunk 1]
```diff
@@ -16,15 +16,15 @@ function handle(e, fn, args) {
 
 	if (out === false) {
 		e.preventDefault();
-		e.stopPropagation();
+               	e.stopPropagation();
```
Comment (leeoniya) [inline src/view/patchEvent.js]: plz reset original 2-tab indentation here.

**Comment by leeoniya** [comment]
this is great, thank you!

a couple notes:

- move `dist/ie8.js` to `/src/view/addons/ie8.js`
- drop the additional ie8 demos and `/demos/ie.js`; i'd prefer not to keep two versions or adapt the existing ones just to have ie8 support.
- i'm still not sure if `e || window.event` is necessary, i don't recall ie8 having problems passing the event into the handlers. is this because the ie8 event shim only affects the global `window.event` prototype?
- i'd prefer to exclude all the compiled builds in `/dist` from pull requests, i will recompile and test them myself (also for performance regressions). sorry, this is mostly my fault for having git track them.

**Comment by e1ectronic** [comment]
Re: `e || window.event`

Internet Explorer 8 doesn't pass event object to event handler if the event handler is assigned via "onXXXX" property. Consider the following example:

`<!DOCTYPE html>
<head><meta http-equiv="X-UA-Compatible" content="IE=8"></head>
<body>
<button id="b1" type="button">onclick</button>
<button id="b2" type="button">attachEvent</button>
<script>
function click(e) { alert(e) }
document.getElementById('b1').onclick = click;  // e = undefined
document.getElementById('b2').attachEvent('onclick', click); // e = event
</script>
</body>`

Clicking the first button would result in 'undefined' message.

Right now `bindEv(el, type, fn)` (patchEvent.js) doesn't use `attachEvent`, that's why event handlers in IE8 don't receive event object as a parameter.

**Comment by leeoniya** [comment]
hmm, yeah it looks like there's no good workaround, `e || window.event` will have to stay.

**Comment by marvinhagemeister** [comment]
Although, I don't have to support IE8 anymore, I just wanted to chime in and give mad respect to you @e1ectronic . It's not an easy task to support such an old engine 👍

**Comment by leeoniya** [comment]
b9ascklxdv0y0

**Comment by e1ectronic** [comment]
Re: moving `ie8.js` to `/src/view/addons/ie8.js`

Could you please edit the build script so that this file and its minified version finally go to the `/dist` folder?
I still don't feel comfortable with package.json magic...

By the way, I added one more polyfill to `ie8.js`: `Object.create`. This function is not supported natively in IE8, and is used in `patch` function from `/addons/patch.js` (known as `node.patch`).

**Comment by leeoniya** [comment]
> Could you please edit the build script so that this file and its minified version finally go to the /dist folder?

i wasn't planning on minifying it, but letting it just live there similar to `src/view/addons/ViewClass.js` which is also meant for `<script>` inclusion.

**Comment by leeoniya** [comment]
hey @e1ectronic were you planning on working on getting this into shape for a merge?

if you don't have time for this soonish, i can do the remaining changes myself, but I want to make sure you're given proper credit for your work and I also want all ie8 work to go in as 1 or 2 commits (yours & my fixup). for me to make 1 commit out of all these (squash) would remove you as the author. i would still credit you in the commit message through. if that would be sufficient, let me know and you can get on with your life :)

alternatively, you could do the squash & rebase this pull request so i can merge it and add a followup commit afterwards.

thanks!

**Comment by e1ectronic** [comment]
Hi Leon, is it looking good now?

**Comment by leeoniya** [comment]
almost!

i made an ie8 branch [1] that has all your changes squashed into a single commit [2]. this is what i'd like to get merged. if you can rebase/squash all your commits to get one commit that matches my squash, that would be great. your git log should only show one unified commit, maybe call it "IE8 support". you would then need to force-push it up to github [3].

if this is too many Git gymnastics for you, you can simply delete your fork, re-fork this repo and make a single commit that matches [2] called "IE8 support".

[1] https://github.com/leeoniya/domvm/commits/ie8
[2] https://github.com/leeoniya/domvm/commit/62d50bf7cb7889555e1369fab9fc9238314d0889
[3] http://stackoverflow.com/questions/5509543/how-do-i-properly-force-a-git-push

**Comment by e1ectronic** [comment]
Thanks for the "easy way"! :) 
New pull request: https://github.com/leeoniya/domvm/pull/137

**Pull request closed without merging**